src/Microsoft.Data/PrimitiveDataFrameColumnContainer.cs

```diff
@@ -0,0 +1,160 @@
+                    Buffers.Add(curBuffer);
+                }
+                curBuffer.Append(values[i]);
+                Length++;
+            }
+        }
+        public PrimitiveDataFrameColumnContainer(IEnumerable<T> values)
+        {
+            values = values ?? throw new ArgumentNullException(nameof(values));
+            if (Buffers.Count == 0)
+            {
+                Buffers.Add(new DataFrameBuffer<T>());
+            }
+            var curBuffer = Buffers[Buffers.Count - 1];
+            foreach (var value in values)
+            {
+                if (curBuffer.Length == curBuffer.MaxCapacity)
+                {
+                    curBuffer = new DataFrameBuffer<T>();
+                    Buffers.Add(curBuffer);
+                }
+                curBuffer.Append(value);
+                Length++;
+            }
+        }
+        public PrimitiveDataFrameColumnContainer() { }
+        public long Length;
+        //TODO:
+        public long NullCount => throw new NotImplementedException();
+        private int GetArrayContainingRowIndex(ref long rowIndex)
+        {
+            if (rowIndex > Length)
+            {
+                throw new ArgumentException($"Index {rowIndex} cannot be greater than the Column's Length {Length}");
+            }
+            int curArrayIndex = 0;
+            int numBuffers = Buffers.Count;
+            while (curArrayIndex < numBuffers && rowIndex > Buffers[curArrayIndex].Length)
+            {
+                rowIndex -= Buffers[curArrayIndex].Length;
+                curArrayIndex++;
+            }
+            return curArrayIndex;
+        }
+
+        public IList<T> this[long startIndex, int length]
+        {
+            get
+            {
+                var ret = new List<T>(length);
+                long endIndex = startIndex + length;
+                int arrayIndex = GetArrayContainingRowIndex(ref startIndex);
+                bool temp = Buffers[arrayIndex][(int)startIndex, length, ret];
+                while (ret.Count < length)
+                {
+                    arrayIndex++;
+                    temp = Buffers[arrayIndex][0, length - ret.Count, ret];
+                }
+                return ret;
+            }
+        }
+        public T this[long rowIndex]
+        {
+            get
+            {
+                int arrayIndex = GetArrayContainingRowIndex(ref rowIndex);
+                return Buffers[arrayIndex][(int)rowIndex];
+            }
+            set
+            {
+                int arrayIndex = GetArrayContainingRowIndex(ref rowIndex);
+                Buffers[arrayIndex][(int)rowIndex] = value;
+            }
+        }
+
+        public override string ToString()
+        {
+            StringBuilder sb = new StringBuilder("");
+            foreach (var buffer in Buffers)
+            {
+                sb.Append(buffer.ToString());
+                // Can this run out of memory? Just being safe here
```

Comment:
StringBuilder can hold int.MaxValue chars unless otherwise specified. ToString is really for debugging, so a limited size string is nice from that point of view..